### PR TITLE
[docs] - Asset config [CON-89]

### PR DESCRIPTION
**Pull request by erinkcochran87**
### Summary & Motivation

This PR updates the docs to include an example of asset configuration using `config_schema`.

[Preview](https://dagster-git-asset-config-docs-elementl.vercel.app/master/concepts/assets/software-defined-assets#context-and-configuration)

### CODE REVIEW

**Review by smackesey — COMMENTED** [review]
I have a few recommended tweaks:

- It should be as clear as possible that the machinery of asset context/config is literally identical to that of ops-- the asset is just wrapping an op. Because config and context are not "advanced" in light of this, I think they need to be front-loaded more in the page. I would make this the last subsection of "Defining assets" (under "Assets with Dependencies")
- I don't like "Context and configuration" as title because there's a little bit of an asymmetry-- configuration is a step below context, and you demonstrate resources (on the same level as config) but don't include that in the title. I think this should actually just mirror the Op concept page and be split into two separate sections: "Asset context" and "Asset config".

That said I've been somewhat out of the loop on the master plan for asset docs restructuring and don't want to stomp on anything, so @sryza what do you think of those proposals?

Otherwise looks good!

File: docs/content/concepts/assets/software-defined-assets.mdx [hunk 1]
```diff
@@ -409,6 +409,20 @@ def my_asset(context):
     return context.resources.api.fetch_table("my_asset")
 ```
 
+Like ops, configuration is also supported for assets. Configuration is accesible through the asset context at runtime and can be used to specify behavior. Note that asset configuration behaves the same as configuration for ops.
```
Comment (sryza) [inline docs/content/concepts/assets/software-defined-assets.mdx:412]: nitpick: accessible spelling

File: docs/content/concepts/assets/software-defined-assets.mdx [hunk 2]
```diff
@@ -398,9 +398,9 @@ def downstream_asset(upstream):
 
 In this case, `ins={"upstream": AssetIn("upstream_asset")}` declares that the contents of the asset with the key `upstream_asset` will be provided to the function argument named `upstream`.
 
-### Using context in assets
+### Context and configuration
 
-Since a software-defined asset contains an op, all of the typical functionality of an op (like the use of resources) is available to an asset.
+Since a software-defined asset contains an op, all the typical functionality of an op - like the use of [resources](/concepts/resources) - is available to an asset. Supplying the `context` parameter provides access to system information for the op, for example:
```
Comment (sryza) [inline docs/content/concepts/assets/software-defined-assets.mdx:403]: maybe mention configuration in this first paragraph as well?

**Review by sryza — COMMENTED** [review]
> I don't like "Context and configuration" as title because there's a little bit of an asymmetry-- configuration is a step below context, and you demonstrate resources (on the same level as config) but don't include that in the title. I think this should actually just mirror the Op concept page and be split into two separate sections: "Asset context" and "Asset config".

I don't feel super strongly opinionated in any direction, but it could make sense to just call the heading "configuration and resources"?

File: docs/content/concepts/assets/software-defined-assets.mdx (hunk 1 above)
Comment (erinkcochran87) [reply docs/content/concepts/assets/software-defined-assets.mdx:412]: Not a nitpick at all, thanks for catching this!

**Comment by smackesey** [comment]
>Could you clarify what you mean here, specifically why that would need to be included in the section title? The original copy for asset context ([here](https://docs.dagster.io/concepts/assets/software-defined-assets#using-context-in-assets)) uses resources as an example and doesn't do this.

Here's the way I'm thinking of it: there is a hierarchical relationship between context (upper) and resources/config (lower). Imagine you were writing an article on animals and you had a section called "Dogs and Golden Retrievers", and in that section you went into equal detail on Golden Retrievers (config) and German Shepherds (resources). Wouldn't "Dogs and Golden Retrievers" feel off as a title then?

>I'm fine with changing this (and the example) to be something more general, like actually accessing system info for the asset. Perhaps that's the way to go instead. Thoughts?

My rec remains as stated in initial review: I think this should actually just mirror the Op concept page and be split into two separate sections: "Asset context" (where resources illustrate context) and "Asset config".

That said, it's really NBD and having given my feedback, feel free to use your judgment.

**Review by smackesey — APPROVED** [review]

**Pull request merged**